$MirOS: ports/print/cups/patches/patch-backend_parallel_c,v 1.1.7.1 2005/03/18 15:51:00 tg Exp $
--- backend/parallel.c.orig	Mon Jan  3 20:29:22 2005
+++ backend/parallel.c	Wed Oct 19 19:01:20 2005
@@ -641,14 +641,11 @@ list_devices(void)
 
   for (i = 0; i < 3; i ++)
   {
-    sprintf(device, "/dev/lpt%d", i);
-    if ((fd = open(device, O_WRONLY)) >= 0)
-    {
-      close(fd);
-      printf("direct parallel:%s \"Unknown\" \"Parallel Port #%d (interrupt-driven)\"\n", device, i + 1);
-    }
+    /* The lpt devices have problems under MirOS / OpenBSD, so we do not
+     * include them.
+     */
 
-    sprintf(device, "/dev/lpa%d", i);
+    snprintf(device, sizeof(device), "/dev/lpa%d", i);
     if ((fd = open(device, O_WRONLY)) >= 0)
     {
       close(fd);
